app/models/pages/page.rb

```diff
@@ -35,6 +37,31 @@ def data
       @data ||= Pages::Data.new
     end
 
+    def parent
+      pathname = Pathname.new(path)
+
+      loop do
```

Comment:
Open ended loops are best avoided. A for loop with a break (even with a high number like 100 loops) is much preferable.